### DISCUSSION TITLE
[Suggestion] Stop Claiming Near Other Claims!

### DISCUSSION THREAD

**Discussion by iZeusi** [GriefPrevention/GriefPrevention#505] [Ideas]
Wagwan.

Wondering if you could add a feature to the plugin that would allow server owners to enable a protection radius on claims (configurable block amount) that stops others from claiming directly next to already claimed land. Example:

Claimed Land = [ ]          New Claimed Land = L
[ ] [ ] [ ] [ ] L
[ ] [ ] [ ] [ ] L
[ ] [ ] [ ] [ ] L
[ ] [ ] [ ] [ ] L

We want to stop this ^^
So instead...
Claimed Land = [ ]          New Claimed Land = L       Space = O
[ ] [ ] [ ] [ ] O O O L
[ ] [ ] [ ] [ ] O O O L
[ ] [ ] [ ] [ ] O O O L
[ ] [ ] [ ] [ ] O O O L

Cheers!

**Comment by TomLewis** [comment]
yeah, get your players to claim more land, not just their builds. 
This isn't a technical issue, it a teaching issue. Put in place rules to not build next to others claims, have a process to report them and teach your users to claim more land than they need to protect their borders.

**Comment by iZeusi** [comment]
I'm sorry, and I'm aware this hasn't come from the developer from the plugin, but this simply wasn't the response I excepted to receive. Imagine having a rule to "Not break our spawn area" instead of using world guard. Do you think that it would go well in most cases? I have rules, yes. However I have found that stopping the players breaking the rules through the use of a plugin, as in, physically stopping them from doing an action, is way more efficient for staff members, who can then spend their time on protecting the server from other rule-breakers, like people who use Xray and mine a lot of ore. Nobody would be forcing you to use this feature if it was added, but it would be a huge help for my server, and I suspect many servers that are currently being run by like-minded server owners.
Cheers, Zeus

**Comment by bigpresh** [comment]
Well, players can claim an area of land they want protected; if they *want* a larger area protected, then claim a larger area.

I think a "not claimed, but still protected" boundary around claims would be nothing but confusing to people.

Consider, for instance, two friends building together, who claim up against each other.  With your proposal, friend B's claim would have to start a certain distance from friend A's claim.  Now, who can build in that protected "no-mans' land" in between them?  Friend A?  Friend B?  Neither?  That just leads to puzzlement, in my book.

It's something that could probably be supported in a GP plugin if you wanted to have someone write something, I suspect, but I think it's a feature that doesn't belong in the core because it's already easy to protect the amount of land you want protected, and that new mechanic would just lead to confusion ("why can't I build here, it's not in a claim!").

**Comment by iZeusi** [comment]
This scenario would require a simple explanation to players, or further, a simple fix. 
The explanation is that ANYBODY, that includes player A and B, can build in that border. Now if that's not enough, and you wish for players to still build next to each other, each player could be given a command to turn off this outer land protection. This protection only covers claims, not building. I dislike that people can grief the outside of a claim, even if it is a large claim, and I dislike that people can taunt and attract players out of their claims for PvP purposes.

**Comment by bigpresh** [comment]
I'm afraid I really don't see it - it'd just make it more complicated for players to understand why they can't claim up against each other, and allowing building in it would just mean people could be griefed, which they wouldn't if they just made their claim bigger.

I know of plenty of players on my server who co-operate and claim right up to each other, and having to figure out that they need to use a command to disable that weird no-mans-land border would just add confusion.

Pretty much all players, though, understand "the land you claim is protected", and that if you want to protect extra land, you just have a bigger claim.

So - I can see that it may be useful to you, and it could well be implemented as a plugin, but I don't think it belongs in GP core, as far as I'm concerned (disclaimer: it isn't my decision to make)

**Comment by TomLewis** [comment]
This is such a stupid idea.
its simple, like I said, if players want to claim land, they claim it. 
You could give them more claim blocks if you wanted, or sell them for in-game currency, or whatever mate, you could have a quest or trade to swap for claim blocks.

**Comment by RoboMWM** [comment]
This is an addon suggestion, until we find a new home for them (if we even do that) they stay open for now.

**Comment by rcmaehl** [comment]
I mainly see this being an issue in shopping districts, in which players try to [explict]block each other so their shop gets the money, and for trolling in general.

My suggestions on possible implementations are as follows in order of potential complexity;

- Wall/Hole blacklisting
    - Prevent the any portion of perimeter of a claim from being walls (blocks surrounded by air) or a hole (air surrounded by blocks)
- Road perimeters
    - Similar to previous implementation but more restrictive
    - Add a border around any claim that is restricted to building at ground level, or up to configurable Y level, only.
        - Prevents players from blocking access to other players claims while still allowing roads and other transport to be built
- Plot Merging
    - By default `claims.mode` is used to determine who can build in the area between plots
    - Allow players to "merge" plots to bypass this
        - Both players have to agree to merge
        - Both players can now build in the merged area, but not each-others plots
- District Claims
    - Allow setting an admin claim as a district
    - Players can claim land in the defined district
    - By default the admin claims permissions is used to determine who can build in the area between plots
    - Comparable to [this wiki section](https://github.com/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#restricting-where-claims-may-be-created) without the overkill of allowing players to grief eachother
    - Players are NOT able to merge plots

**Comment by rcmaehl** [comment]
This is also a copy of #124 👀

**Comment by RoboMWM** [comment]
Thanks for pointing that out, will close in favor of that one.

(Interestingly, that was the issue I used in the FAQ).

**Comment by RoboMWM** [comment]
duplicate of #124
